src/ripple/basics/strHex.h

```diff
@@ -62,22 +66,33 @@ charUnHex (char c)
 }
 /** @} */
 
-// NIKB TODO cleanup this function and reduce the need for the many overloads
-//           it has in various places.
-template<class FwdIt>
-std::string strHex (FwdIt first, int size)
+template<class InputIterator>
+std::string strHex(InputIterator begin, InputIterator end)
+{
+    std::string result;
+    result.reserve (std::distance(begin, end));
+    boost::algorithm::hex (begin, end, std::back_inserter(result));
+    return result;
+}
+template <class T>
```

Comment:
Unfortunately, we need to constrain this template or we `strHex(10)` will resolve to this overload instead of `uint64_t` overload. I'd like to stay away from a trait if we can (I think they're hard to read). What do you think of ```
template <class T, class = decltype(std::declval<T>().begin())>
```
To constrain this? (I change the `cbegin` to `begin`).

Reply:
Right, that makes sense. I thought the change of that call site to use `charHex` was the right thing to do, but like you pointed out, that is not quite right. So, that puts us into the boat of constraining this function template taking in some `T`. 

Your constraint works. Personally, I find traits much more composable, allow reuse, and more readable. But, YMMV. There isn't much TMP or heavy use of constraints to make my argument strong in this codebase.

Reply:
If this check was doing any boolean logic - i.e. check if `begin` and `end` exist, then I'd use a trait. If there was an existing library of traits, I might add this to it and use that trait. But for a one-off check of a single member function I find the terse syntax clearer.

BTW, I considered adding another overload for `int` to solve this problem. That should work too, but I think I'd rather just use the simple constraint.